### PR TITLE
Support object interpolation

### PR DESCRIPTION
**Pull request by tjhiggins**
Make consumers an array of strings
Add test of parameter with object default

### CODE REVIEW

File: src/dependency-manager/src/schema/spec/component-spec.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export class ParameterDefinitionSpec {
   description?: string;
 
   @IsOptional()
-  @JSONSchema(AnyOf('boolean', 'number', 'string', 'null'))
+  @JSONSchema(AnyOf('boolean', 'number', 'object', 'string', 'null'))
   default?: boolean | number | string | null;
```
Comment (dan-barrett) [inline src/dependency-manager/src/schema/spec/component-spec.ts:68]: We should add `object` to the `default` typescript type here too.

File: src/dependency-manager/src/schema/spec/resource-spec.ts [hunk 2]
```diff
@@ -47,8 +47,8 @@ export class VolumeSpec {
   readonly?: boolean | string;
 }
 
-export type EnvironmentSpecValue = boolean | null | number | string;
-export const EnvironmentDictiory = DictionaryOfAny('boolean', 'null', 'number', 'string');
+export type EnvironmentSpecValue = boolean | null | number | object | string;
```
Comment (dan-barrett) [inline src/dependency-manager/src/schema/spec/resource-spec.ts:50]: Likewise, we should add `array` to the `EnvironmentSpecValue ` typescript type too.

Comment (tjhiggins) [reply src/dependency-manager/src/schema/spec/resource-spec.ts:50]: arrays are objects

File: src/dependency-manager/src/schema/spec/transform/resource-transform.ts [hunk 3]
```diff
@@ -48,7 +48,12 @@ export const transformResourceSpecEnvironment = (environment: Dictionary<Environ
     if (v === undefined || v === null) {
       continue;
     }
-    output[k] = `${v}`;
+
+    if (v instanceof Object) {
```
Comment (dan-barrett) [inline src/dependency-manager/src/schema/spec/transform/resource-transform.ts:52]: Also I don't think this is sufficient to check if it's an array. And I'm assuming we'll also want to stringify arrays, correct?

Comment (tjhiggins) [reply src/dependency-manager/src/schema/spec/transform/resource-transform.ts:52]: it is

**Pull request merged**